### PR TITLE
Add licensing support

### PR DESCRIPTION
**Pull request by sergiusens**
(Closes LP: #1527453)

### CODE REVIEW

File: integration_tests/test_license.py [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+#
+# You should have received a copy of the GNU General Public License
+# along with this program.  If not, see <http://www.gnu.org/licenses/>.
+
+import os
+
+from testtools.matchers import (
+    DirExists,
+    FileExists,
+    FileContains,
```
Comment (elopio) [inline integration_tests/test_license.py]: FileContains before FileExists, so it's alphabetical.

File: snapcraft/tests/test_meta.py [hunk 2]
```diff
@@ -317,6 +334,41 @@ def test_create_meta_with_vararg_config(self, mock_the_open,
...
+            ),
+        ])
+
+        self.mock_copyfile.assert_has_calls([
+            call('my-icon.png', os.path.join(self.meta_dir,
+                 'my-icon.png')),
+            call('file.apparmor', os.path.join(self.meta_dir,
+                 'file.apparmor')),
+            call('file.seccomp', os.path.join(self.meta_dir,
+                 'file.seccomp')),
```
Comment (elopio) [inline snapcraft/tests/test_meta.py]: These three calls are not related to the license. You can use mock.ANY instead, I think here we only care about the license call.

File: snapcraft/tests/test_meta.py [hunk 3]
```diff
@@ -317,6 +334,41 @@ def test_create_meta_with_vararg_config(self, mock_the_open,
...
+
+        mock_the_open.assert_has_calls(self.expected_open_calls)
+        mock_wrap_exe.assert_has_calls([
+            call(
+                '$SNAP_APP_PATH/bin/bash',
+                os.path.join(os.path.abspath(os.curdir),
+                             'snap/bin/bash.wrapper'),
+                args=None, shebang=None,
+            ),
+        ])
```
Comment (elopio) [inline snapcraft/tests/test_meta.py]: How is this assertion related to the license? You might be testing too much in this test, this seems to me like something to do in a separate test.

File: snapcraft/tests/test_yaml.py [hunk 4]
```diff
@@ -542,6 +542,38 @@ def test_invalid_part_name_plugin_raises_exception(self):
...
+
+    def test_license_version_without_license_raises_exception(self):
+        self.data['license-version'] = '1.1'
+
+        with self.assertRaises(snapcraft.yaml.SnapcraftSchemaError) as raised:
+            snapcraft.yaml._validate_snapcraft_yaml(self.data)
+
+        expected_message = "'license' is a dependency of 'license-version'"
+        self.assertEqual(raised.exception.message, expected_message,
+                         msg=self.data)
```
Comment (elopio) [inline snapcraft/tests/test_yaml.py]: What about a scenario for version without agreement?

**Comment by elopio** [comment]
You have here some files without the trailing new line. I think we should adopt the practice of ending all files with a new line.
Also I left some comments for discussion.

**Comment by sergiusens** [comment]
@chipaca maybe you want to take a look.

File: snapcraft/tests/test_yaml.py (hunk 4 above)
Comment (sergiusens) [inline snapcraft/tests/test_yaml.py]: added `def test_license_with_license_version(self):`

**Comment by sergiusens** [comment]
@elopio I think I gone through all your requests.

**Comment by chipaca** [comment]
:+1:

**Comment by elopio** [comment]
yes, thanks, :+1:
